worker/worker.go

```diff
@@ -199,6 +199,12 @@ type (
 		// The logger is the only optional parameter. Defaults to the noop logger. The Run ID and Workflow ID used during replay are derived
 		// from execution.
 		ReplayWorkflowExecution(ctx context.Context, service workflowservice.WorkflowServiceClient, logger log.Logger, namespace string, execution workflow.Execution) error
+
+		// GetWorkflowResult gets the result from a succesfully replayed workflow that finished with WorkflowExecutionCompleted.
+		// WorkflowID is optional and defaults to ReplayId.
+		//
+		// Note: For all replay functions if no WorkflowID is given for replay the default is ReplayId.
+		GetWorkflowResult(workflowID string, valuePtr interface{}) error
```

Comment:
I am not quite sure we are ready for exposing workflow results from our replayers. We have intentionally avoided doing this in other replayers. I think we need more discussion around the API. For example, does this show errors from a workflow? How do I know if it didn't complete?

I do understand the desire for this and how it helps the replay tests. Options:

1. Get together as a team and decide that we want workflow results provided as a result of replay (includes success and fail instead of just wft fail like today)
2. Remove this from the interface and you can leave this on the internal struct for now and just cast to that internal struct in these tests
3. Assert in your workflow and panic on failure

Reply:
> ready for exposing workflow results from our replayers.
Not sure what conditions would make us ready?

> show errors from a workflow
We already expose this on replay with the current replay API

> How do I know if it didn't complete?
What didn't complete the original workflow history or the replayed workflow? If the original workflow didn't complete you know that (you have the history) and the replayed workflow can't complete. If the original did, but the replayed didn't we fail.

Reply:
> Not sure what conditions would make us ready?

Just confirming group wide that we want to across SDKs and if there are any caveats here

> If the original workflow didn't complete you know that (you have the history)

By this logic you also have the history for workflow completion so we don't need to return the result. If we're gonna return result of a fully-complete workflow from the replayer, the user should be able to know the difference between workflow incomplete vs workflow complete successfully vs workflow complete unsuccessfully vs workflow task failure.

> Currently if a WFT fails on replay we expose the error. If the original workflow failed I think we just exit and don't check it matched with the replay which seems like a bug?

It's not a bug in today's approach because our replayer does not expose anything about workflow complete (success or failure). But since you are adding workflow complete information, we need to provide full workflow complete information, not only success.

Reply:
>By this logic you also have the history for workflow completion so we don't need to return the result. If we're gonna return result of a fully-complete workflow from the replayer, the user should be able to know the difference between workflow incomplete vs workflow complete successfully vs workflow complete unsuccessfully vs workflow task failure.

Your missing the point of what I am saying. I am saying for most part they can. Except for workflow execution failed.

>It's not a bug in today's approach because our replayer does not expose anything about workflow complete (success or failure). But since you are adding workflow complete information, we need to provide full workflow complete information, not only success.

It isn't about exposing it is about. I am saying we don't check it at all. That isn't consistent with how we handle Workflow success or continue as new.

Reply:
Ok, maybe I am misunderstanding. To ask it simply: Why are we exposing `GetWorkflowResult` in workflow replayer today in Go SDK and not in any other SDK? Also: Since we're now returning workflow result, how can I as a user differentiate replayer failure from workflow failure?

Reply:
Discussed offline, will drop from the interface for know and revisit exposing more completely later